CleverTapSDKTests/CleverTapInstanceTests.m

```diff
@@ -70,4 +74,134 @@ - (void)test_clevertap_get_global_instance {
+}
+
+- (void)testShouldDropEventReturnsNOForFetchType {
+    NSDictionary *event = @{CLTAP_EVENT_NAME: @"anyEvent"};
+    BOOL result = [self.cleverTapInstance _shouldDropEvent:event withType:CleverTapEventTypeFetch];
+    XCTAssertFalse(result, @"Fetch event type should never be dropped");
+}
+
+- (void)testShouldDropEventDropsSystemEventWhenOptedOutAndDisallowSystemEvents {
+    NSDictionary *event = @{CLTAP_EVENT_NAME: @"_system_event_name"};
+    self.cleverTapInstance.currentUserOptedOut = YES;
+    self.cleverTapInstance.currentUserOptedOutAllowSystemEvents = NO;
+
+    id validatorMock = OCMClassMock([CTValidator class]);
+    OCMStub([validatorMock isRestrictedEventName:@"_system_event_name"]).andReturn(YES);
+
+    BOOL result = [self.cleverTapInstance _shouldDropEvent:event withType:123]; // any non-fetch type
+    XCTAssertTrue(result, @"System event should be dropped if opted out and system events disallowed");
+
+    [validatorMock stopMocking];
+}
+
+- (void)testShouldDropEventDoesNotDropSystemEventWhenAllowSystemEventsIsYES {
+    NSDictionary *event = @{CLTAP_EVENT_NAME: @"_system_event_name"};
+    self.cleverTapInstance.currentUserOptedOut = YES;
+    self.cleverTapInstance.currentUserOptedOutAllowSystemEvents = YES;
+
+    id validatorMock = OCMClassMock([CTValidator class]);
+    OCMStub([validatorMock isRestrictedEventName:@"_system_event_name"]).andReturn(YES);
+
+    BOOL result = [self.cleverTapInstance _shouldDropEvent:event withType:123];
+    XCTAssertFalse(result, @"System event should not be dropped if system events are allowed");
+
+    [validatorMock stopMocking];
+}
+
+- (void)testShouldDropEventDropsCustomEventWhenOptedOut {
+    NSDictionary *event = @{CLTAP_EVENT_NAME: @"custom_event"};
+    self.cleverTapInstance.currentUserOptedOut = YES;
+    self.cleverTapInstance.currentUserOptedOutAllowSystemEvents = NO;
+
+    id validatorMock = OCMClassMock([CTValidator class]);
+    OCMStub([validatorMock isRestrictedEventName:@"custom_event"]).andReturn(NO);
+
+    BOOL result = [self.cleverTapInstance _shouldDropEvent:event withType:123];
+    XCTAssertTrue(result, @"Custom event should be dropped if user opted out");
+
+    [validatorMock stopMocking];
+}
+
+- (void)testShouldNotDropEventIfUserNotOptedOut {
+    NSDictionary *event = @{CLTAP_EVENT_NAME: @"custom_event"};
+    self.cleverTapInstance.currentUserOptedOut = NO;
+
+    id validatorMock = OCMClassMock([CTValidator class]);
+    OCMStub([validatorMock isRestrictedEventName:@"custom_event"]).andReturn(NO);
+
+    BOOL result = [self.cleverTapInstance _shouldDropEvent:event withType:123];
+    XCTAssertFalse(result, @"Event should not be dropped if user not opted out");
+
+    [validatorMock stopMocking];
+}
+
+- (void)testShouldDropEventWhenMuted {
+    NSDictionary *event = @{CLTAP_EVENT_NAME: @"any_event"};
+    self.cleverTapInstance.currentUserOptedOut = NO;
+
+    id instanceMock = OCMPartialMock(self.cleverTapInstance);
```

Comment:
Getting error "Another mock is already associated with object CleverTap.test (NSInternalInconsistencyException)" on:
`id instanceMock = OCMPartialMock(self.cleverTapInstance);`

Call `stopMocking` in the other methods where the instance is mocked.

Reply:
done